tests/e2e/test_sampling_params.py

```diff
@@ -0,0 +1,323 @@
+import pytest
+from spyre_util import get_spyre_backend_list, get_spyre_model_list
+from vllm import LLM, SamplingParams
+
+
+@pytest.fixture(scope="module")
+@pytest.mark.parametrize("model", get_spyre_model_list())
+def spyre_model(model: str) -> LLM:
+    return LLM(model=model)
+
+
+@pytest.mark.parametrize("backend", get_spyre_backend_list())
+def test_spyre_batch1_temperature(spyre_model: LLM, backend: str,
+                                  monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setenv("VLLM_SPYRE_DYNAMO_BACKEND", backend)
+    monkeypatch.setenv("VLLM_SPYRE_OVERRIDE_SIGNALS_HANDLER", "1")
+
+    prompt = "The capital of the United Kingdom is"
+    params = SamplingParams(temperature=0.0, seed=8780, max_tokens=5)
+
+    output1 = spyre_model.generate(prompt, params)[0]
+    output2 = spyre_model.generate(prompt, params)[0]
+
+    assert output1.outputs[0].text == output2.outputs[0].text
+    assert "London" in output1.outputs[0].text
+
+
+@pytest.mark.parametrize("backend", get_spyre_backend_list())
+def test_spyre_batch1_max_tokens(spyre_model: LLM, backend: str,
+                                 monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setenv("VLLM_SPYRE_DYNAMO_BACKEND", backend)
+    monkeypatch.setenv("VLLM_SPYRE_OVERRIDE_SIGNALS_HANDLER", "1")
+
+    prompt = "Count to twenty"
+    params = SamplingParams(temperature=0, seed=8780, max_tokens=15)
+
+    output = spyre_model.generate(prompt, params)[0]
+
+    assert len(output.outputs[0].token_ids) == 15
+
+
+@pytest.mark.parametrize("backend", get_spyre_backend_list())
+def test_spyre_batch1_stop_sequence(spyre_model: LLM, backend: str,
+                                    monkeypatch: pytest.MonkeyPatch):
+    monkeypatch.setenv("VLLM_SPYRE_DYNAMO_BACKEND", backend)
+    monkeypatch.setenv("VLLM_SPYRE_OVERRIDE_SIGNALS_HANDLER", "1")
+
+    stop_str = "London"
+    prompt = f"The best way to travel from Paris to {stop_str} is by train."
+    params = SamplingParams(stop=[stop_str], max_tokens=50)
```

Comment:
We should also test without stop and make sure that the finish_reason is different.